### PR TITLE
PR to fix issue #1304 with cosmic's new identifier and cosmic url

### PR DESCRIPTION
**Pull request by hassanfa**
This PR fixes handling new COSMIC id tag and updates the URL for COSMIC from ../id= (direct id) to ../q= (query/search).

This will help resolve four cases:
1. Older tags sometimes get merged or moved to new ids. This causes some tags with ../id= stub URL not work properly and end up not being found.
2. New tags won't work with ../id=, since COSMIC's stub URL seems to be always resolving to COSM.
3. New tags and old tags will be stored as string, NOT int.
4. COSMIC tag in INFO field is also parses the same way as CSQ with `&` separator.

**How to test**:
1. how to install it
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [x] tests executed by @hassanfa 
- [ ] "merge and deploy" approved by

### CODE REVIEW

**Comment by hassanfa** [comment]
Supposed to be draft, I'll request for review when it is finished. Need to make a new demo file with new COSMIC identifiers.

**Comment by moonso** [comment]
Hi @hassanfa , best thing would be if you perform the tests yourself and upload the picture(s). Then someone else can focus on the actual code review

**Comment by northwestwitch** [comment]
Hi @hassanfa , how is it going with this PR, do you need help?

**Comment by hassanfa** [comment]
Oh no! I thought I finished this. 😣 Fixing it right now. Thanks for the reminder

**Comment by hassanfa** [comment]
While testing this I noticed COSV tag does not work with Scout's link generation.

Example: https://cancer.sanger.ac.uk/cosmic/mutation/overview?id=21711103

COSM: COSM**10704** and COSV: COSV**52662048**

URL for COSM:  https://cancer.sanger.ac.uk/cosmic/mutation/overview?id=10704
URL for COSV:  https://cancer.sanger.ac.uk/cosmic/mutation/overview?id=52662048

The second URL doesn't work. 

Also, we store COSMIC id's int part only:

https://github.com/Clinical-Genomics/scout/blob/80c127e27887539156c83c84ac640b0c67a50070/scout/parse/variant/transcript.py#L219-L220

I propose to use stub URL for query instead:

URL for COSM:  https://cancer.sanger.ac.uk/cosmic/search?q=COSM10704
URL for COSV:  https://cancer.sanger.ac.uk/cosmic/search?q=COSV52662048

Both of them work and will end up on same page.

So example:

https://github.com/Clinical-Genomics/scout/blob/2bbcd012a1122ebaee4990da9952d4d0013dca69/scout/server/links.py#L375-L378

This way we still link to older loaded data (those that are int, and go to id= stub), and those will be processed as string and go to q= stub url

thoughts?

**Comment by northwestwitch** [comment]
> I propose to use stub URL for query instead:
> 
> URL for COSM: https://cancer.sanger.ac.uk/cosmic/search?q=COSM10704
> URL for COSV: https://cancer.sanger.ac.uk/cosmic/search?q=COSV52662048
> 
> Both of them work and will end up on same page.

Sounds like a good solution to me!

**Comment by northwestwitch** [comment]
Let me know when you are done so I can review!

**Comment by hassanfa** [comment]
Altered 2 variant VCF with COSV and COSM. Created two case: master and cosmic_cosv_com same VCF. Attached here as well.

Test 1: Variants are loaded into db as str:
image

Test 2: URL created successfully:
COSM:
image
COSV:
image

Test 3: Proof master fails to parse COSV and only processes COSM:
image

And URL for int one is created successfully:
image


Attached are yaml files for these two cases and their VCF.
[yaml_files.zip](https://github.com/Clinical-Genomics/scout/files/4386397/yaml_files.zip)
[cancer_test_cosmic_new_id.vcf.gz](https://github.com/Clinical-Genomics/scout/files/4386399/cancer_test_cosmic_new_id.vcf.gz)

**Comment by hassanfa** [comment]
@northwestwitch it is ready, but I managed to drop coverage due to parsing COSMIC tag in INFO field. Updating CHANGELOG as well.

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,9 @@ About changelog [here](https://keepachangelog.com/en/1.0.0/)
 ### Fixed
 - Fix the link to Ensembl for SV variants when genome build 38.
 - Arrange information in columns on variant page
+- Fix so that new cosmic identifier is also acceptable #1304
```
Comment (northwestwitch) [inline CHANGELOG.md]: Could you add here which one it is, for reference when we do the next release?

File: scout/parse/variant/transcript.py [hunk 2]
```diff
@@ -216,7 +216,7 @@ def parse_transcripts(raw_transcripts, allele=None):
             for variant_id in variant_ids.split("&"):
                 if variant_id.startswith("rs"):
                     transcript["dbsnp"].append(variant_id)
-                elif variant_id.startswith("COSM"):
-                    transcript["cosmic"].append(int(variant_id[4:]))
+                elif variant_id.startswith("COSM") or variant_id.startswith("COSV"):
```
Comment (northwestwitch) [inline scout/parse/variant/transcript.py]: 👍

File: scout/parse/variant/variant.py [hunk 3]
```diff
@@ -249,7 +249,8 @@ def parse_variant(
 
     cosmic_tag = variant.INFO.get("COSMIC")
     if cosmic_tag:
-        cosmic_ids.add(cosmic_tag[4:])
+        for cosmic_id in cosmic_tag.split("&"):
```
Comment (northwestwitch) [inline scout/parse/variant/variant.py]: 👍

**Comment by northwestwitch** [comment]
Naa it's says coverage remained the same! :)

**Comment by northwestwitch** [comment]
Shall we merge @hassanfa ?

**Comment by hassanfa** [comment]
Yes, please merge. I clarified new tag in CHANGELOG.

**Comment by northwestwitch** [comment]
> Yes, please merge.

Aren't you allowed to merge? You should have the permission, right?

**Comment by hassanfa** [comment]
I'm allowed to, but I didn't know I should. I'll merge as soon as it is green all over the board.

**Pull request merged**